api/app/signals/apps/services/domain/filescanner.py

```diff
@@ -0,0 +1,44 @@
+# SPDX-License-Identifier: MPL-2.0
+# Copyright (C) 2021 Gemeente Amsterdam
+"""
+Check file types, both for fresh uploads and stored files.
+"""
+import mimetypes
+
+import magic
+
+ALLOWED_MIME_TYPES = ['image/jpeg', 'image/png', 'image/gif']
+
+
+class BadFileError(Exception):
```

Comment:
Do we need just one `BadFileError` exception or do we want several exceptions that are easily catchable in `try except` blocks without looking at the message. For example `FileRejected`, `UnsupportedFileType`, `UnsupportedFileExtension`, `TypeExtensionMismatch` (just spitballing a couple of thoughts here).

The can also live in an `exceptions.py` file.

Reply:
I added more specific exceptions, they are also tested at the services level